views/view_0_send_sms_http.class.php

```diff
@@ -1,168 +1,298 @@
+          break;
+        case 'person':
+        default:
+          if (!empty($_POST['personid'])) {
+            $recips = $GLOBALS['system']->getDBObjectData('person', Array('id' => $_POST['personid'], '!mobile_tel' => '', '!status' => 'archived'), 'AND');
+            $blanks = $GLOBALS['system']->getDBObjectData('person', Array('id' => $_POST['personid'], 'mobile_tel' => '', '!status' => 'archived'), 'AND');
+            $archived = $GLOBALS['system']->getDBObjectData('person', Array('id' => $_POST['personid'], 'status' => 'archived'), 'AND');
+            $GLOBALS['system']->includeDBClass('person');
+          }
+          break;
+      }
+    }
+    if (!empty($recips)) {
+      foreach ($recips as $recip) {
+        $mobile_tels[$recip['mobile_tel']] = 1;
+      }
+      $mobile_tels = array_keys($mobile_tels);
+    }
 
-		if (empty($recips)) {
-			print_message("Did not find any recipients with mobile numbers.  Message not sent.", 'error');
-		} else {
+    return Array($recips,$blanks,$archived,$mobile_tels);
+  }
 
-			$mobile_tels = array();
-			foreach ($recips as $recip) {
-				$mobile_tels[$recip['mobile_tel']] = 1;
-			}
-			$mobile_tels = array_keys($mobile_tels);
+  function sendSMS($message, $recips, $mobile_tels)
+  {
+    $response = '';
+    $success = false;
+    $content = SMS_HTTP_POST_TEMPLATE;
+    $content = str_replace('_USER_MOBILE_', urlencode($GLOBALS['user_system']->getCurrentUser('mobile_tel')), $content);
+    $content = str_replace('_USER_EMAIL_', urlencode($GLOBALS['user_system']->getCurrentUser('email')), $content);
+    $content = str_replace('_MESSAGE_', urlencode($message), $content);
+    $content = str_replace('_RECIPIENTS_COMMAS_', urlencode(implode(',', $mobile_tels)), $content);
+    $content = str_replace('_RECIPIENTS_NEWLINES_', urlencode(implode("\n", $mobile_tels)), $content);
+    if(defined('SMS_RECIPIENT_ARRAY_PARAMETER')){
+        $content = str_replace('_RECIPIENTS_ARRAY_', SMS_RECIPIENT_ARRAY_PARAMETER . '[]=' . implode('&' . SMS_RECIPIENT_ARRAY_PARAMETER . '[]=', $mobile_tels), $content);	
+    }    
+			
+    $header  = "" . SMS_HTTP_HEADER_TEMPLATE;
+    $header  = $header . "Content-Length: ".strlen($content)."\r\n"
+                       . "Content-Type: application/x-www-form-urlencoded\r\n";
 
-			$message = array_get($_POST, 'message');
-			// Known issue: if their session was timed out when they entered the message,
-			// the message is not propagated after they submit the login form, so they'll get the
-			// "message is empty" error here.
-			if (empty($message) || strlen($message) > SMS_MAX_LENGTH) {
-				print_message("Your message is empty or too long", "error");
-				return;
-			}
+    $opts = Array(
+      'http' => Array(
+        'method'        => 'POST',
+        'content'       => $content,
+        'header'        => $header
+      )
+    );
+    // To work with HTTP Server errors ourselves, override the system error_handler
+    set_error_handler(null);
```

Comment:
Thanks for that.  Yes, this PR is closed but I just thought I'd use this venue for the discussion so there is context.  I'll work on merging #213 soon.